### DISCUSSION TITLE
Mesh EKS add-ons services

### DISCUSSION THREAD

**Discussion by arnaud-s** [linkerd/linkerd2#11527] [Help] [answered]
Hi,

I have EKS cluster deployed using Terraforme EKS module https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest.

I successfully installed Linkerd using the helm charts, and managed to mesh all my services in the cluster.

Except the add-ons that were installed when deploying the cluster (coredns, kube-proxy, csi-driver,) basically everything under the `kube-system` namespace.

I tried to update the addons and add the pod annotation via AWS CLI but it doesn't mesh the service.

Did I miss something? Is there a way to mesh these services?

Thanks for your help!

**Comment by mateiidavid** [comment, marked as answer]
Hi, we generally don't recommend meshing services under `kube-system`. This is recommended [in the Kubernetes docs](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#avoiding-operating-on-the-kube-system-namespace) ("admission webhooks shouldn't run in `kube-system`"). We also cover this in our [HA docs](https://linkerd.io/2.14/features/ha/#exclude-the-kube-system-namespace)

**Reply by arnaud-s** [reply]
Thanks for your reply!
